src/language-js/print/ternary.js

```diff
@@ -284,28 +285,45 @@ function printTernary(path, options, print, args) {
     hasComment(consequentNode, CommentCheckFlags.Dangling)
   ) {
     path.call((childPath) => {
-      consequentComments.push(
-        printDanglingComments(childPath, options),
-        hardline,
-      );
+      const danglingComments = printDanglingComments(childPath, options, {
+        marker: experimentalTernaryDanglingCommentMarker,
+      });
+      if (danglingComments) {
+        consequentComments.push(danglingComments, hardline);
+      }
```

Comment:
I took some time to find the root cause, I think the problem here is we are printing the dangling comments in the `consequent` and the `consequent` will also print them.

Do you think we can check node type of `consequent` instead?

Something like:

```js
const nodesShouldPrintDanglingComments = node =>
  node.type === 'ArrayExpression' || 
  node.type === 'ObjectExpression'
if (!nodesShouldPrintDanglingComments (node)) {
  ...
}
```

Reply:
This sounds plausible to me fwiw